File: 이진영/Stack/프로세스.js
function solution(priorities, location) {
    let map = new Map();
    let q = [];
    let count = 0;
    
    // 인덱스 - 우선 순위 map
    for (let i = 0; i < priorities.length; i++) {
        q.push(i);
        map.set(i, priorities[i]);
    }
    
    while (q.length > 0) {
        // 제일 높은 우선 순위 체크
        let max = Math.max(...q.map(i => priorities[i]));
        
        // 제일 높지 않으면 뒤로 보냄 => 맨 앞에 제일 높은 우선순위의 인덱스 위치
        while (priorities[q[0]] !== max) {
            q.push(q.shift());
        }

        // count 증가, 맨 앞 인덱스 끝내기
        count++;
        if (q.shift() === location) return count;
    }
}

File: 정민선/프로그래머스/프로세스.py
from collections import deque
def solution(priorities, location):
    answer = 0
    indexPriorities = deque()
    for i, p in enumerate(priorities):
        indexPriorities.append((p, i))
        
    while indexPriorities:
        now = indexPriorities.popleft()
        if any(now[0] < item[0] for item in indexPriorities):
            indexPriorities.append(now)
        else:
            answer += 1
            if now[1] == location:
                return answer
            
priorities = [2, 1, 3, 2]
solution(priorities, location=2)